src/machine/machine_gameboyadvance.go

```diff
@@ -54,7 +54,7 @@ func (d FramebufDisplay) Size() (x, y int16) {
 }
 
 func (d FramebufDisplay) SetPixel(x, y int16, c color.RGBA) {
-	d.port[y][x].Set(uint16(c.R)&0x1f | uint16(c.G)&0x1f<<5 | uint16(c.B)&0x1f<<10)
+	d.port[y][x].Set(((uint16(c.R) >> 3) & 31) | (((uint16(c.G) >> 3) & 31) << 5) | (((uint16(c.B) >> 3) & 31) << 10))
```

Comment:
The colors here are `uint8`, so after you shift right by 3 the `& 31` is unnecessary. It will probably be optimized away anyway, but the code would perhaps be cleaner by leaving the mask out.

Not a big issue, only worth fixing if you think it's worth it.

Reply:
Indeed the bit mask is useless here, I just modified it in following commit.

I tested the code with both emulators [mGBA](https://mgba.io/) and [VisualBoyAdvance-M](https://vba-m.com/). I don't have any physical GBA to test it further.